graylog2-web-interface/templates/index.html.js

```diff
@@ -0,0 +1,44 @@
+const fs = require('fs');
+const vendorModule = JSON.parse(fs.readFileSync('build/vendor-module.json', 'utf8'));
+
+function linkTagForCssFile(cssFile) {
+  return '<link href="' + cssFile + '" rel="stylesheet">';
+}
+
+function scriptTagForJsFile(jsFile) {
+  return '<script src="' + jsFile + '"></script>';
+}
+
+module.exports = function (templateParams) {
+  const htmlWebpackPlugin = templateParams.htmlWebpackPlugin;
+  const manifest = htmlWebpackPlugin.files.manifest ? ' manifest="' + htmlWebpackPlugin.files.manifest + '"' : '';
+  const title = htmlWebpackPlugin.options.title ? htmlWebpackPlugin.options.title : 'Webpack App';
+  const favIcon = htmlWebpackPlugin.files.favicon ? '<link rel="shortcut icon" href="' + htmlWebpackPlugin.files.favicon + '">' : '';
+
+  const cssFiles = htmlWebpackPlugin.files.css.map(function(css) {
+    return linkTagForCssFile(htmlWebpackPlugin.files.css[css]);
+  }).join("\n");
+
+  const vendorFiles = Object.keys(vendorModule.files.chunks).map(function(chunk) {
+    return "\t" + scriptTagForJsFile('/' + vendorModule.files.chunks[chunk].entry);
+  });
+  const jsFiles = Object.keys(htmlWebpackPlugin.files.chunks).map(function(chunk) {
+    return "\t" + scriptTagForJsFile(htmlWebpackPlugin.files.chunks[chunk].entry);
+  }).join("\n");
+
+  return '\
+  <!DOCTYPE html>\n\
```

Comment:
Same here, it looks clumsy. The difference is that now we are looking up the name of the vendor bundle (which is part of a different compiler run, so it is not available before the actual template is rendered) dynamically (because it's now being a assigned a chunkhash as part of the name), therefore we need to use a JS template instead of a static one.